gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinClusterManager.java

```diff
@@ -264,6 +268,12 @@ public synchronized void start() {
     this.eventBus.register(this);
     this.multiManager.connect();
 
+    // Standalone mode registers a handler to clean up on leadership change, so don't do the cleanup
+    // now even if the option to clean up on startup is set.
+    if (this.cleanUpJobsOnStartup && !this.isStandaloneMode) {
```

Comment:
Is this check needed for correctness or to avoid duplicate clean up calls? If it is the latter, shouldn't the 2nd call be handled as a No-op?

Reply:
It is to avoid duplicate clean up calls. The code would be correct without this check. A standalone mode manager can transition through multiple leadership change events and there is only one clean up call per transition, so there is no need to dedupe/no-op in that code path.